flat_mutation_reader.cc

```diff
@@ -1344,6 +1345,8 @@ flat_mutation_reader downgrade_to_v1(flat_mutation_reader_v2 r) {
         }
         virtual future<> next_partition() override {
             clear_buffer_to_next_partition();
+            // The client didn't receive the paired-up range tombstone change and isn't interested.
```

Comment:
Someone coming to this code without any knowledge about what problem was solved will have trouble understanding what you mean here.

The comment lacks context: range tombstone change paired-up to what?
Also, the client does not receive range tombstone _changes_, they receive range tombstones (in v1 format). It's best to be accurate with terminology.
Also, there may have been no range tombstone change at all so there's no pairing up to be done.
Let's say it's understood why the client isn't interested the range tombstone - but why do we need to reset the assembler in this case? It's not obvious.

It would be best to explain what the problem is and when it happens. It happens conditionally: *If* we have an active range tombstone - which means we "started" a range tombstone but didn't finish it yet, and the client calls next_partition at this point, *then* it means we don't need to push the range tombstone so we can discard any information about an active range. And we need to explicitly discard it because the assembler requires there to not be any active range tombstone when it finishes.

Reply:
I would also change the comment a bit as this is not definitive that there is any range tombstone change at all so 'may have not received' might be more appropriate.

Reply:
I elaborated in the comments.